dataflow/src/org/checkerframework/dataflow/cfg/CFGBuilder.java

```diff
@@ -757,6 +781,47 @@ public void popFrame() {
             labels.add(exitLabel);
             return labels;
         }
+
+        @Override
+        public String toString() {
+            StringBuilder sb = new StringBuilder();
+            sb.append("TryStack: exitLabel: " + this.exitLabel + '\n');
+            if (this.frames.isEmpty()) {
+                sb.append("No TryFrames.\n");
+            }
+            for (TryFrame tf : this.frames) {
+                sb.append(tf.toString());
+            }
+            return sb.toString();
+        }
+    }
+
+    /**
+     * A map that keeps track of new labels added within a try block. For names that are outside of
+     * the try block, the finally label is returned. This ensures that a finally block is executed
+     * when control flows outside of the try block.
+     */
+    @SuppressWarnings("serial")
+    protected static class FakeLabelsMap extends HashMap<Name, Label> implements Map<Name, Label> {
```

Comment:
`FakeLabelsMap` is probably not the best name.
Is `FinallyScopeLabelMap` better? Other suggestions?

Reply:
`FinallyScopeLabelMap ` is better.  What about `TryFinallyLablesMap`?   I don't have a preference.